documentation/behind-the-scenes-of-adding-a-file.md

```diff
@@ -0,0 +1,233 @@
+## Purpose
+
+The steps outlined below are taken to "add a file with Filecoin Pin".  This document is intended to provide more info about what happens "behind the scenes" as it uses underlying libraries like [`synapse`](glossary.md#synapse) and the [Filecoin Onchain Cloud](glossary.md#filecoin-onchain-cloud) offering.
+
+## Diagram
+
+_Key_ 
+Blue | Orange | FP | SP
+-- | -- | -- | --
+non-blockchain step | blockchain step | Filecoin Pin | Storage Provider
```

Comment:
I seem to remember some convo when I first joined where we were attempting to change "storage provider" to "service provider".. not sure if that's still the case.

```suggestion
Blue | Orange | FP | SP
-- | -- | -- | --
non-blockchain step | blockchain step | Filecoin Pin | Service Provider
```

Reply:
Yeah, you're right.  I don't know why I did Storage Provider throughout this document.  I changing to Service Provider.